src/QsCompiler/CompilationManager/Diagnostics.cs

```diff
@@ -27,15 +27,22 @@ invalidFragmentEnding switch
                 _ => throw new ArgumentException("Unrecognized fragment ending."),
             };
 
-        private static DiagnosticSeverity Severity(QsCompilerDiagnostic msg)
+        private static DiagnosticSeverity Severity(QsCompilerDiagnostic msg, HashSet<int>? warningsAsErrors)
         {
             if (msg.Diagnostic.IsError)
             {
                 return DiagnosticSeverity.Error;
             }
             else if (msg.Diagnostic.IsWarning)
             {
-                return DiagnosticSeverity.Warning;
+                if (warningsAsErrors != null && warningsAsErrors.Contains(msg.Code))
```

Comment:
Could use the null conditional operator here:

`if (warningsAsErrors?.Contains(msg.Code) == true)`